dotcom-rendering/src/web/components/StickyVideo.importable.tsx

```diff
@@ -91,27 +92,64 @@ const stickyContainerStyles = (height: number) => css`
 interface Props {
 	isPlaying: boolean;
 	children: React.ReactNode;
+	videoId: string;
 }
 
-export const StickyVideo = ({ isPlaying, children }: Props) => {
+export const StickyVideo = ({ isPlaying, children, videoId }: Props) => {
 	const [isSticky, setIsSticky] = useState(false);
+	const [isClosed, setIsClosed] = useState(false);
+
 	const [isIntersecting, setRef] = useIsInView({
 		threshold: 0.5,
 		repeat: true,
 	});
 
 	useEffect(() => {
 		setIsSticky(isPlaying && !isIntersecting);
+		setIsClosed(false);
 	}, [isIntersecting, isPlaying]);
 
+	useEffect(() => {
+		if (isSticky) {
+			submitComponentEvent({
+				component: {
+					componentType: 'STICKY_VIDEO',
+					id: videoId,
+				},
+				action: 'STICK',
+			});
+		} else if (isClosed) {
+			submitComponentEvent({
+				component: {
+					componentType: 'STICKY_VIDEO',
+					id: videoId,
+				},
+				action: 'CLOSE',
+			});
+		} else {
+			submitComponentEvent({
+				component: {
+					componentType: 'STICKY_VIDEO',
+					id: videoId,
+				},
+				action: 'RETURN',
+			});
+		}
+	}, [isSticky, isClosed, videoId]);
```

Comment:
Does the `videoId` change in the component? Looking at the code, it's set once, and never changes.

Reply:
I think eslint would complain if we removed it from the dependencies array?

Reply:
Yeah, it's static but the linter asks for it. I think it's also nice to have it listed in the dependency array in terms of declaring what is being used but maybe this is just me looking for silver linings!

Reply:
Ah, it's the exhaustive deps. Definitely has a lot of gain, so nice to have it. Thanks for explaining!